File: Application_Source/main/java/com/example/go/myapplication03/AESCipher.kt
package com.example.go.myapplication03

import java.nio.charset.StandardCharsets
import javax.crypto.Cipher
import javax.crypto.spec.IvParameterSpec
import javax.crypto.spec.SecretKeySpec
import java.security.MessageDigest
import android.util.Base64

class AESCipher(key: String) {
    private val secretKey: SecretKeySpec

    init {
        val sha256 = MessageDigest.getInstance("SHA-256")
        val hashedKey = sha256.digest(key.toByteArray())
        secretKey = SecretKeySpec(hashedKey, "AES")
    }

    fun encrypt(message: String): String {
        val cipher = Cipher.getInstance("AES/CBC/PKCS5Padding")
        cipher.init(Cipher.ENCRYPT_MODE, secretKey, IvParameterSpec(ByteArray(16)))
        val encrypted = cipher.doFinal(message.toByteArray(StandardCharsets.UTF_8))
        return Base64.encodeToString(encrypted, Base64.DEFAULT)
    }

    fun decrypt(enc: String): String {
        val cipher = Cipher.getInstance("AES/CBC/PKCS5Padding")
        cipher.init(Cipher.DECRYPT_MODE, secretKey, IvParameterSpec(ByteArray(16)))
        val decrypted = cipher.doFinal(Base64.decode(enc, Base64.DEFAULT))
        return String(decrypted, StandardCharsets.UTF_8)
    }
}

File: blockchain/AESCipher.py
import base64
import hashlib
from Crypto.Cipher import AES

class AESCipher(object):
    def __init__(self, key):
        self.key = hashlib.sha256(key.encode()).digest()
        
    def pad(self, message):
        block_size = AES.block_size
        padding_size = block_size - len(message) % block_size
        padding = chr(padding_size).encode() * padding_size
        padded_message = message + padding
        return padded_message
    
    def encrypt(self, message):
        raw = self.pad(message.encode())
        cipher = AES.new(self.key, AES.MODE_CBC, self.__iv().encode('utf8'))
        enc = cipher.encrypt(raw)
        return base64.b64encode(enc).decode('utf-8')
    
    def decrypt(self, enc):
        enc = base64.b64decode(enc)
        cipher = AES.new(self.key, AES.MODE_CBC, self.__iv().encode('utf8'))
        dec = cipher.decrypt(enc)
        return self.unpad(dec).decode('utf-8')
    
    def __iv(self):
        return chr(0) * 16
    
    def unpad(self, message):
        padding_size = message[-1]
        return message[:-padding_size]
